script/main.js

```diff
@@ -75,30 +71,35 @@ async function showNotebook(){
 
 }
 async function deleteNote(){
-
-    const response = await fetch(`${APIUrl}/delete`,{
-        method: 'DELETE',
-        headers: {
-            'Content-type':'application/json'
-        },
-        body: JSON.stringify({id: valueInput})
-    })
-
-    if (!response.ok){
-        throw new Error('There is a problem.');
-    }
-    const result = await response.json();
-    console.log(`The result is: ${result}`)
+    const deleteButtons = document.querySelectorAll('.fa-trash-alt');
+
+    deleteButtons.forEach(button => {
+        button.addEventListener('click', async function (event) {
```

Comment:
An event listener in a loop can bring weird errors because its called multiple times, the event listener should trigger the loop. that way its only called once.

Reply:
fixed